src/app/shared/components/Breadcrumbs/BreadcrumbsItem/BreadcrumbsItem.tsx

```diff
@@ -121,31 +145,249 @@ const BreadcrumbsItem = (props: BreadcrumbsItemProps): JSX.Element => {
 
   const ItemIconComponent = iconService.getItemIcon(true);
 
+  const findCurrentFolder = (currentBreadcrumb) => {
+    const foldersList: DriveItemData[] = [];
+
+    for (const itemsInAllitems in allItems) {
+      const selectedFolder = allItems[itemsInAllitems].find(
+        (item) => item.id === currentBreadcrumb.id && item.name === currentBreadcrumb.name,
+      );
+      if (selectedFolder) foldersList.push(selectedFolder as DriveItemData);
+    }
+    return foldersList;
+  };
```

Comment:
I think that not exists the possibility to finde more than one item with same id and name. 
If so, an alternative to the findCurrentFolder implementation that saves traversing the entire list and returns a single item, not an array.
```suggestion
  const findCurrentFolder = (currentBreadcrumb) => allItems.find((item) => item.id === currentBreadcrumb.id && item.name === currentBreadcrumb.name);
```

Reply:
I do it this way because of the type of object (allItems) that contains arrays, only with the 'find' it doesn't work

Reply:
ah okay! 
In that case if you are only going to return one item at most I think it would be better to return the value and not an array with the item in it. 
Or it is any reason to store it in an array?

Reply:
Yes, it is the type of data expected in moveItemsToTrash, downloadItemsThunk, setItemsToMove

Reply:
Okay, but that's no reason to choose that data type for the function to return. 
A function is usually better to return the type according to the expected behaviour. In this case, if we expect it to return a single item (the current folder), it should return the data type that corresponds to that single item, and not a list with one item. 
To send it as an argument through the functions you mention you can add the list when you are going to pass it `[item]`. 
I'm sorry if I insist a lot on something that doesn't seem important, but it's to keep the code as consistent and clean as possible.  🙏

Reply:
Maybe I have not explained well or it is not clear but the function is created to reuse the code in these cases.
If I change the function to return a result that then I have to manipulate in several places in the same way I am repeating code to generate the expected result, I think it is better to do it only once inside the function and get the expected result.
But if you continue thinking that it is better that the result of the function is not an array I change it. 😉